cmd/mock/rbac.go

```diff
@@ -0,0 +1,136 @@
+	    "last": "/api/v1/(resources)/?offset=90&limit=10"
+	  },
+	  "data": [
+	    {
+	      "permission": "cost-management:*:read",
+	      "resourceDefinitions": [
+	        {
+	          "attributeFilter": {
+	            "key": "cost-management.aws.account",
+	            "operation": "equal",
+	            "value": "123456"
+	          }
+	        }
+	      ]
+	    }
+	  ]
+	}
+*/
+type RbacAccessRequest struct {
+	Application string `query:"application"`
+	Username    string `query:"username"`
+	OrderBy     string `query:"order_by"`
+	Limit       int    `query:"limit"`
+	Offset      int    `query:"offset"`
+}
+
+type RbacMeta struct {
+	Count int `json:"count"`
+}
+
+type RbacLinks struct {
+	First    string `json:"first,omitempty"`
+	Previous string `json:"previous,omitempty"`
+	Next     string `json:"next,omitempty"`
+	Last     string `json:"last,omitempty"`
+}
+
+type RbacData struct {
+	Permission          string `json:"permission,omitempty"`
+	ResourceDefinitions struct {
+		AttributeFilter struct {
+			Key       string `json:"key,omitempty"`
+			Operation string `json:"operation,omitempty"`
+			Value     string `json:"value,omitempty"`
+		} `json:"attributeFilter,omitempty"`
+	} `json:"resourceDefinitions,omitempty"`
+}
+
+type RbacAccessResponse struct {
+	Meta  RbacMeta   `json:"meta"`
+	Links RbacLinks  `json:"links,omitempty"`
+	Data  []RbacData `json:"data"`
+}
+
+func MockRbac(c echo.Context) error {
+	var request RbacAccessRequest
+	if err := c.Bind(&request); err != nil {
+		c.Error(err)
+		return err
+	}
+
+	const xRhIdentityHeader = "X-Rh-Identity"
+	var (
+		xRhIdentityB64  string
+		xRhIdentityJson []byte
+		xRhIdentity     identity.Identity
+		err             error
+	)
+
+	if xRhIdentityB64 = c.Request().Header.Get(xRhIdentityHeader); xRhIdentityB64 == "" {
+		return c.JSON(http.StatusBadRequest, fmt.Sprintf("'%s' header cannot be empty", xRhIdentityHeader))
+	}
+	if xRhIdentityJson, err = b64.StdEncoding.DecodeString(xRhIdentityB64); err != nil {
+		return c.JSON(http.StatusBadRequest, fmt.Sprintf("'%s' wrong base64 format", xRhIdentityHeader))
+	}
+	if err = json.Unmarshal(xRhIdentityJson, &xRhIdentity); err != nil {
+		return c.JSON(http.StatusBadRequest, fmt.Sprintf("'%s' wrong json format", xRhIdentityHeader))
+	}
+
+	outputAdmin := RbacAccessResponse{
+		Data: []RbacData{
+			{
+				Permission: "content-sources:*:read",
+			},
+			{
+				Permission: "content-sources:*:write",
+			},
+		},
+	}
+	outputDefault := RbacAccessResponse{
+		Data: []RbacData{
+			{
+				Permission: "content-sources:*:read",
+			},
+		},
+	}
+	outputEmpty := RbacAccessResponse{
+		Data: []RbacData{},
+	}
+
+	switch {
+	case xRhIdentity.OrgID == "12345" && xRhIdentity.AccountNumber == "11111":
+		return c.JSON(http.StatusOK, outputAdmin)
+	case xRhIdentity.OrgID == "12345" && xRhIdentity.AccountNumber == "22222":
+		return c.JSON(http.StatusOK, outputDefault)
+	default:
+		return c.JSON(http.StatusOK, outputEmpty)
+	}
```

Comment:
could you base this off orgId instead of account number?    I'd probably suggest allow all for most all OrgIds, but have some more restricted ones for testing:  

```
switch {
	case xRhIdentity.OrgID == "12345" && xRhIdentity.AccountNumber == "readonly":
		return c.JSON(http.StatusOK, outputDefault)
	case xRhIdentity.OrgID == "12345" && xRhIdentity.AccountNumber == "noperms":
		return c.JSON(http.StatusOK, outputEmpty)
	default:
		return c.JSON(http.StatusOK, outputAdmin)
	}
```

I think this will make things simpler to develop with (especially when working in the UI where you don't control the orgId).  (also i get that orgId is typically a number, but i think it'd work as a string?)

Reply:
sure! thanks for that! I didn't see from that point of view!